projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -44,6 +46,7 @@ export class AuthService {
     protected authStorageService: AuthStorageService,
     protected authRedirectService: AuthRedirectService,
     protected routingService: RoutingService,
+    protected winRef: WindowRef,
```

Comment:
I am not sure if we are able to squeeze this PR for 6.0, as today there is a code freeze. So based on that for the next minor such adding new dependency into the constructor will be a breaking change.

Another questions is, do we need to add this dependency here? What if we move setting new item into localStorage to `oauth-lib-wrapper.service.ts`? The `winRef` dependency is already there, we can just extend `initLoginFlow` method reponsibility.

Reply:
yes i was thinking about this, because auth service is extended by asm-auth-service, and adding dependency here, requires adding this dependency to asm-service, so i was thinking about handling this local storage setup in oauth-lib-wrapper. But after all it seems for me that oauth-wrapper is just a wrapper without specific additional logic and as for me auth-service seems to be more suitable place for additional logic.

But i didn't think about breaking changes, in that case, i agree that we should get rid of this dep. and use oauth-lib-wrapper for this

Reply:
Zmienione